### PR TITLE
Filing year

### PR DESCRIPTION
**Pull request by wpears**
Depends on #279 

Closes https://github.com/cfpb/hmda-platform-ui/issues/269

Reworks how filings are modeled in the store and fires requests to the `filings` endpoint to populate it along with institutions.

### CODE REVIEW

**Comment by awolfe76** [comment]
@wpears - I'm seeing `TypeError: Cannot read property 'institutionId' of undefined(…)` when loading the institutions page. One of 2 may load and others not. It seems to be inconsistent.

(I've gone into the Keycloak admin and added my user to `0,1,2,3`.)

**Comment by awolfe76** [comment]
@wpears - just a general question. I still see we're calling `/institutions` to get the full list, ok there. But then we're calling `/institutions/<institution>`  and `/institutions/<institution>/filings/<period>`. 

Do we still need the call to `/institutions/<institution>`? I was hoping we could remove it because the other 2 endpoints get us everything we need. _I think_.

**Comment by wpears** [comment]
Nope, the filings endpoint doesn't have institution name, which is only returned from the institutions endpoint.

**Comment by awolfe76** [comment]
Right, the `/institutions` has it. So is there anything that `/institutions/<institution>` is getting us?

**Comment by wpears** [comment]
`/institutions/<institution>` allows us to discover the available filing periods for each institution, so we can request only the `/institutions/<institution>/filings/<filing>` that matches the current filing period.

We talked a bit about whether there would always be a default filing for each period, etc, but right now as it stands, just constructing filings endpoints from the info in `/institutions` would result in 404s for institutions that don't have filings for the current filing period.

An alternate approach would be to just intercept 404s and assume they indicate that no filing exists for that institution in the current filing period, but that seems a bit heavy-handed to me.

**Comment by awolfe76** [comment]
I think the 404 is an ok response, especially given that scenario is very unlikely in the future because, like you said, we'll have the default filing period there. That would save us from additional API calls too.

File: src/js/actions/index.js [hunk 1]
```diff
@@ -448,13 +461,50 @@ export function fetchEachInstitution(institutions) {
...
  * Fetch an institution via the api and dispatch an action with the results
  */
 export function fetchInstitution(institution) {
   return dispatch => {
     dispatch(requestInstitution())
     return getInstitution(institution.id)
-      .then(json => dispatch(receiveInstitution(json)))
+      .then(json => {
+        dispatch(receiveInstitution(json))
+          console.log('filings for dispatch', json,json.filings)
```
Comment (awolfe76) [inline src/js/actions/index.js:486]: Will you still need the logging here?

File: src/js/api.js [hunk 2]
```diff
@@ -23,7 +23,11 @@ function sendFetch(suffix, options = {method: 'GET'}){
 
   console.log('fetching from', url, 'with options', fetchOptions)
   return fetch(url, fetchOptions)
-    .then(response => {console.log('respose from fetch', response);return response.json()})
+    .then(response => {
+      const json = response.json()
+      console.log('respose from fetch', response, json)
```
Comment (awolfe76) [inline src/js/api.js:28]: Logging again

File: src/js/components/Institutions.jsx [hunk 3]
```diff
@@ -90,30 +90,32 @@ const renderRefile = (code, institutionId, period) => {
...
+  for(let i=0; i<institutions.length; i++){
+    if(institutions[i].id === filing.institutionId) return institutions[i]
+  }
+  return null
+}
+
 export default class Institution extends Component {
   render() {
-    var self = this
+    console.log('inst filings',this.props.institutions, this.props.filings)
```
Comment (awolfe76) [inline src/js/components/Institutions.jsx:102]: Logging

**Review by awolfe76 — APPROVED** [review]
Just the few comments about logging.

**Pull request merged**